tests/test_adapters.py

```diff
@@ -57,66 +75,145 @@ async def test_prompt_adapter():
+    base_model_name = "/granite/granite-3b-base-v2/step_75000_ckpt"
+    adapter_name = "granite-3b-code-instruct-lora"
+    lora_path = str(FIXTURES_DIR) + "/" + adapter_name
+    request = BatchedGenerationRequest(
+        adapter_id=adapter_name,
+        model_id=base_model_name,
+    )
+    mock_lora_requests = PropertyMock(
+        side_effect=[
+            [],
+            [
+                LoRARequest(
+                    lora_name=adapter_name,
+                    lora_int_id=0,
+                    lora_path=lora_path,
+                ),
+            ],
+            [
+                LoRARequest(
+                    lora_name=adapter_name,
+                    lora_int_id=0,
+                    lora_path=lora_path,
+                ),
+            ],
+        ]
+    )
+    type(vllm_model_handler).lora_requests = mock_lora_requests
+    adapter_store = AdapterStore(cache_path=FIXTURES_DIR, adapters={})
+
+    adapters_1 = await validate_adapters(request, adapter_store, vllm_model_handler)
+    adapters_2 = await validate_adapters(request, adapter_store, vllm_model_handler)
+    # Ensure the vllm model handler only received a single load request
+    vllm_model_handler.load_lora_adapter.assert_awaited_once_with(
+        request=LoadLoRAAdapterRequest(
+            lora_path=str(FIXTURES_DIR) + "/" + adapter_name,
+            lora_name=adapter_name,
+        ),
+        base_model_name=base_model_name,
+    )
+
+    # Metadata isn't cached locally
+    assert len(adapter_store.adapters) == 0
+    # Same unique ID is reused for the second request
     assert (
         adapters_1["lora_request"].lora_int_id == adapters_2["lora_request"].lora_int_id
     )
 
 
 @pytest.mark.asyncio
-async def test_store_handles_multiple_adapters():
+async def test_store_handles_multiple_adapters(vllm_model_handler):
     adapter_store = AdapterStore(cache_path=FIXTURES_DIR, adapters={})
 
-    adapter_name = "granite-3b-code-instruct-lora"
+    adapter_name = "bloom_sentiment_1"
     request = BatchedGenerationRequest(
         adapter_id=adapter_name,
     )
-    adapters_1 = await validate_adapters(request, adapter_store=adapter_store)
+    adapters_1 = await validate_adapters(request, adapter_store, vllm_model_handler)
 
     adapter_name = "bloomz-560m-prompt-adapter"
     request = BatchedGenerationRequest(
         adapter_id=adapter_name,
     )
-    adapters_2 = await validate_adapters(request, adapter_store=adapter_store)
+    adapters_2 = await validate_adapters(request, adapter_store, vllm_model_handler)
 
     assert len(adapter_store.adapters) == 2
     assert (
-        adapters_1["lora_request"].lora_int_id
+        adapters_1["prompt_adapter_request"].prompt_adapter_id
         < adapters_2["prompt_adapter_request"].prompt_adapter_id
     )
 
 
 @pytest.mark.asyncio
-async def test_cache_handles_concurrent_loads():
+async def test_cache_handles_concurrent_loads(vllm_model_handler):
```

Comment:
For completeness we could do this with lora adapters too just to make sure our locking works so that we only load once